python/whl.bzl

```diff
@@ -40,6 +40,15 @@ def _extract_wheels(ctx, wheels):
     if result.return_code:
         fail("extract_wheels failed: %s (%s)" % (result.stdout, result.stderr))
 
+    patch_runtime = ctx.attr.patch_runtime
+    patch_runtime = [ctx.path(p) for p in patch_runtime]
+    name = ctx.attr.key
```

Comment:
I don't think `<key>/__init__.py` is a valid logic for choosing the file to patch. First, there is not necessarily even a subdirectory in the wheel (e.g. six consists of just six.py), and second, the package name might not match the directory name (e.g. "tensorflow-gpu" is the package name, but top level directory is "tensorflow").
Also, sometimes we might want to patch some other file than `__init__.py`, so I wonder if we could make this more generic. How about instead having a `patch` attribute that would contain patch files, and we'd apply those patches? Then any file extracted from the wheel could be patched, or new ones added. Tensorflow repo uses similar idiom with their `patched_http_archive`. WDYT?

Reply:
Yep, makes more sense indeed.

Reply:
Fixed